### PR TITLE
Bring clickhouse-test changes from private

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

No point on keeping this in private. It's introducing unnecessary diffs and it might be useful for some people.

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 0c7496bedf7d9b0bc51381f7cac81af464bdee48 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60924/0c7496bedf7d9b0bc51381f7cac81af464bdee48/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/clickhouse-test [hunk 1]
```diff
@@ -1441,18 +1629,51 @@ class TestCase:
...
+                # to avoid breaking CSV parser
+                result.description = result.description.replace("\0", "")
+            else:
+                with SharedEngineReplacer(
+                    self.case_file,
+                    args.replace_replicated_with_shared,
+                    args.replace_non_replicated_with_shared,
+                    False,
+                ):
+                    with SharedEngineReplacer(
```
Comment (alexey-milovidov) [inline tests/clickhouse-test:1655]: Looks strange.

Comment (Algunenano) [reply tests/clickhouse-test:1655]: This is just ugly (and to me usually hard to read) python code.  It just triggers thing in its destructor. I'm going to focus of having things passing in both repos with the same code first and then we can refactor things if necessary.

**Comment by Algunenano** [comment]
.

**Pull request merged**